lib/nesting_unique_head.rb

```diff
@@ -0,0 +1,23 @@
+# Nested unique header generation
+require 'middleman-core/renderers/redcarpet'
+
+$headers_history = {}
```

Comment:
I think this doesn't need to be a global, but the class should still be able to access it? Would be nice to avoid using total globals in the unlikely case in clashes with another imported gem or something like that.

I'd also be curious about how this could leak with multiple runnings of the thing. In the other program, we zero this value in `initialize`, but IIRC maybe there were issues with imported partials with `include` in the YAML frontmatter?